guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,396 @@
+Finally, we can put together our pipeline
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=layers, augmentations_per_image=3
+)
+
+"""
+Let's check out the results!
+"""
+
+
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Awesome! As you can see, no images were randomly rotated. You can customize the
+pipeline however you like:
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=[keras_cv.layers.GridMask(), keras_cv.layers.Grayscale(output_channels=3)],
+    augmentations_per_image=1,
+)
+
+"""
+This pipeline will either apply `GrayScale` or GridMask:
+"""
+
+
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Looks great! You can use `RandomAugmentationPipeline` however you want.
+"""
+
+"""
+## Training a CNN
+
+As a final exercise, let's take some of these layers for a spin. In this section, we
+will use `CutMix`, `MixUp`, and `RandAugment` to train a state of the art `ResNet50`
+image classifier on the Oxford flowers dataset.
+
+"""
+
+
+def preprocess_for_model(inputs):
+    images, labels = inputs["images"], inputs["labels"]
+    images = tf.cast(images, tf.float32)
+    return images, labels
+
+
+train_dataset = (
+    load_dataset()
+    .map(apply_rand_augment, num_parallel_calls=AUTOTUNE)
+    .map(cut_mix_and_mix_up, num_parallel_calls=AUTOTUNE)
+)
+
+visualize_dataset(train_dataset, "CutMix, MixUp and RandAugment")
+
+train_dataset = train_dataset.map(preprocess_for_model, num_parallel_calls=AUTOTUNE)
+
+test_dataset = load_dataset(split="test")
+test_dataset = test_dataset.map(preprocess_for_model, num_parallel_calls=AUTOTUNE)
+
+train_dataset = train_dataset.prefetch(5)
+test_dataset = test_dataset.prefetch(5)
+
+"""
+Next we should create a the model itself. Notice that we use `label_smoothing=0.1` in
+the loss function. When using `MixUp`, label smoothing is _highly_ recommended.
+"""
+
+input_shape = IMAGE_SIZE + (3,)
+
+
+def get_model():
+    inputs = keras.layers.Input(input_shape)
+    x = keras.layers.Rescaling(1 / 255.0)(inputs)
+    x = applications.ResNet50(
+        input_shape=input_shape, classes=num_classes, weights=None
+    )(x)
+    model = keras.Model(inputs, x)
+    model.compile(
+        loss=losses.CategoricalCrossentropy(label_smoothing=0.1),
+        optimizer=optimizers.SGD(momentum=0.9),
+        metrics=["accuracy"],
+    )
+    return model
+
+
+"""
+Finally we train the model:
+"""
+
+strategy = tf.distribute.MirroredStrategy()
+with strategy.scope():
+    model = get_model()
+    model.fit(
+        train_dataset,
+        steps_per_epoch=1000,
```

Comment:
I think it's good practice to compute these.

```python
data, ds_info = tfds.load('oxford_flowers102', with_info=True, as_supervised=True)
train_ds, valid_ds, test_ds = data['train'], data['validation'], data['test']

steps_per_epoch=len(train_ds) // BATCH_SIZE
```

Reply:
Done using ds_info.  You can't do len(train_ds) in graph mode - so don't want to recommend this to users.